frontend/catalyst/python_interface/inspection/draw.py

```diff
@@ -15,27 +15,50 @@
 
 from __future__ import annotations
 
+import io
 import warnings
+from collections.abc import Callable
 from functools import wraps
-from typing import TYPE_CHECKING
+from shutil import which
 
 from pennylane.tape import QuantumScript
+from pennylane.workflow.qnode import QNode
+from xdsl.dialects.builtin import ModuleOp
 
+from catalyst.jit import QJIT
+from catalyst.passes.pass_api import PassPipelineWrapper
 from catalyst.python_interface.compiler import Compiler
 
 from .collector import QMLCollector
+from .specs import StopCompilation
 from .xdsl_conversion import get_mlir_module
 
-if TYPE_CHECKING:
-    from pennylane.typing import Callable
-    from pennylane.workflow.qnode import QNode
+HAS_MATPLOTLIB = True
+try:
+    import matplotlib.image as mpimg
+    import matplotlib.pyplot as plt
+except ModuleNotFoundError:
+    HAS_MATPLOTLIB = False
+
+HAS_PYDOT = True
+try:
+    import pydot
+except ModuleNotFoundError:
+    HAS_PYDOT = False
+
+HAS_GRAPHVIZ = True
+if which("dot") is None:
+    HAS_GRAPHVIZ = False
+
+from catalyst.python_interface.inspection.construct_circuit_dag import ConstructCircuitDAG
+from catalyst.python_interface.inspection.pydot_dag_builder import PyDotDAGBuilder
```

Comment:
These can be relative imports
```suggestion
from .construct_circuit_dag import ConstructCircuitDAG
from .pydot_dag_builder import PyDotDAGBuilder
```